### PR TITLE
feat(data-fetchers): migrate bigwig to TS

### PR DESCRIPTION
**Pull request by manzt**
Extends #739 & #740

### CODE REVIEW

File: src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts [hunk 1]
```diff
@@ -0,0 +1,323 @@
+/* TODO: Support esm
```
Comment (manzt) [inline src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts]: Not sure what support ESM means here? This JavaScript file uses the `import` & `export` syntax, which is what I assume is meant by ESM.

File: src/missing-types.d.ts [hunk 2]
```diff
@@ -327,9 +330,13 @@ declare module '@higlass/utils' {
...
         absPosition: number,
-        chrInfo: ChromInfo<string>
+        chrInfo: Pick<ChromInfo, 'cumPositions' | 'chromLengths'>
     ): [chr: string, chrPositon: number, offset: number, insertPoint: number];
-    export function chrToAbs<Name>(chrom: Name, chromPos: number, chromInfo: ChromInfo<Name>): number;
+    export function chrToAbs<Name = string>(
+        chrom: Name,
+        chromPos: number,
+        chromInfo: Pick<ChromInfo<Name>, 'chrPositions'>
+    ): number;
```
Comment (manzt) [inline src/missing-types.d.ts:339]: Unless these function implementations change, the use of `Pick` narrows the type of `ChromInfo` to just the essential fields so they are more reusable.

File: src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts (hunk 1 above)
Comment (sehilyi) [inline src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts]: This was an error. Let's remove the line.

```suggestion
/*
```

File: src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+        bwFile: BigWig | null;
+        TILE_SIZE: number;
+        errorTxt: string;
+        dataPromises: Promise<unknown>[];
+        chromSizes: ChromInfo<string> & { chrToAbs: (name: string, pos: number) => number };
+        assembly?: Assembly;
+        tilesetInfoLoading?: boolean;
+
+        constructor() {
+            // TODO(2022-06-29): shouldn't have hard coded chromSizesUrl
```
Comment (sehilyi) [inline src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts:53]: I should have dealt with the assembly issue with much care. I will address this when I work on supporting custom assembly #582 soon

File: src/missing-types.d.ts (hunk 2 above)
Comment (sehilyi) [reply src/missing-types.d.ts:339]: I agree. Added to a previous PR, but we should use this in other Gosling.js codes (e.g., `assembly.ts`).

**Review by sehilyi — APPROVED** [review]
Looks good to me! 

Did not look at line-by-line changes as GitHub does not properly shows detailed diff between js and ts files, but I tested locally and all demos looks good.

File: src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts (hunk 3 above)
Comment (manzt) [reply src/data-fetcher/bigwig/higlass-bigwig-datafetcher.ts:53]: great!

**Pull request merged**